app/workers/facilities/dental_service_reload_job.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+# require 'facilities/bulk_json_client'
+require 'common/exceptions'
+# require 'facility_access'
+require 'sentry_logging'
+
+module Facilities
+  class DentalServiceError < StandardError
+  end
+
+  class DentalServiceReloadJob
+    include Sidekiq::Worker
+    include SentryLogging
+
+    def fetch_dental_service_data(file)
+      CSV.read(file, headers: true)
+    end
+
+    def update_cache(model, facilities)
+      facilities.each_key do |k|
+        attrs = { station_number: k,
+                  local_updated: Time.now.utc.iso8601 }
+        obj = model.find(k)
+        if obj
+          obj.update(attrs)
+        else
+          model.create(attrs)
+        end
+      end
+    end
+
+    def invalidate_removed(model, facility_keys)
+      invalidate = model.keys - facility_keys
+      invalidate.each { |x| model.delete(x) }
+      logger.info "Removed #{invalidate.size} obsolete entries from cache"
+    end
+
+    def parse_dental_service_data(records)
+      facilities = Hash.new { |h, k| h[k] = {} }
+      records.each do |rec|
+        id = rec['unique_id']
+        facilities[id]
+      end
+      facilities
+    end
+
+    def update_dental_service_data
+      dental_file = Rails.root.join('lib', 'facilities', 'dental_service_data', 'dental_services.csv')
+      records = fetch_dental_service_data(dental_file)
+      facilities = parse_dental_service_data(records)
+      update_cache(FacilityDentalService, facilities)
+      logger.info "Updated facility dental service cache for #{facilities.size} facilities"
+      invalidate_removed(FacilityDentalService, facilities.keys)
+    rescue Facilities::DentalServiceError => e
```

Comment:
I have added a test for sending `Facilities::DentalServiceError` to sentry. This will only be triggered if there is trouble reading the file or converting it to a CSV.

These are all good points and I don't have nearly the grasp you do on what happens when the workers run and especially when an error occurs. I wonder if it would be beneficial to have a tech debt ticket to look into the facilities jobs and what happens when they error. Then after we have createed/modified these jobs we can verify that they are acting appropriately and in the same manner for the purposes of troubleshooting.

Reply:
You're right that it's larger than just this ticket! Another issue sounds like the best option.

Reply:
New issue for facility jobs error handling: https://github.com/department-of-veterans-affairs/vets-contrib/issues/3056